src/django_components/templatetags/component_tags.py

```diff
@@ -57,34 +136,192 @@ def _component_dependencies(type: Literal["js", "css"]) -> SafeString:
             f"Unknown dependency type in {{% component_dependencies %}}. Must be one of 'css' or 'js', got {type}"
         )
 
-    return mark_safe(placeholder)
+    return TextNode(mark_safe(placeholder))
+
+
+@register.tag("component_css_dependencies")
+@with_tag_spec(
+    TagSpec(
+        tag="component_css_dependencies",
+        end_tag=None,  # inline-only
+    )
+)
+def component_css_dependencies(parser: Parser, token: Token, tag_spec: TagSpec) -> TextNode:
+    """
+    Marks location where CSS link tags should be rendered after the whole HTML has been generated.
+
+    Generally, this should be inserted into the `<head>` tag of the HTML.
 
+    If the generated HTML does NOT contain any `{% component_css_dependencies %}` tags, CSS links
+    are by default inserted into the `<head>` tag of the HTML. (See
```

Comment:
Reading this again, are we sure this is the right behavior? Feels very magical that this happens automatically and I'm not sure it outweighs the user not understanding how the framework works.

Reply:
IMO this is a matter of preference, I personally don't mind if a framework does things for me automatically, as long as I can still modify stuff when needed.

Also it feels like there's already quite an upfront cost, requiring 8 steps to fully install the package with JS / CSS support. Which would be 9 with configuring `{% component_dependencies %}`.

If I imagine I'm starting a new project and installing django_components right from the start, then I have no HTML files yet. And so if I'm told to insert `{% component_css_dependencies %}`, that'd be even more steps for me - create the `templates` dir, look up how the base HTML structure looks like (I always google that), load the `component_tags` library, and only finally insert the tags.

So actually I think the default behaviour removes quite a bit friction there.

On the other hand, I think this should be mentioned early on so it's not magical, but just sane defaults.

The installation steps can be split into two parts to make it less overwhelming:
1. Basic installation (5 steps)
2. Adding support for JS and CSS (3 steps)

So I think that as 4th step of the 2nd section, we can mention that "optionally you can use `{% component_dependencies %}` and by default they are inserted here and here".

This is how that looks (the installation page still needs polishing)

<img width="1200" alt="Screenshot 2024-10-20 at 09 08 20" src="https://github.com/user-attachments/assets/57c273d4-3290-40d8-a94e-c1a3d15e5860">

<img width="1207" alt="Screenshot 2024-10-20 at 09 08 09" src="https://github.com/user-attachments/assets/0bba17d0-cb05-4e46-8671-43a356129921">

<img width="1217" alt="Screenshot 2024-10-20 at 09 07 54" src="https://github.com/user-attachments/assets/b79e954c-edd6-43b4-830c-1f8cdf24136d">

Reply:
> Step 4 in the first block of setup is optional too right?

Yup! Then there's also step 5 in the first block, but I just realized it's not in the screenshots 😅